fineract-provider/src/main/java/org/apache/fineract/cob/loan/CheckLoanRepaymentOverdueBusinessStep.java

```diff
@@ -71,4 +79,12 @@ public String getEnumStyledName() {
     public String getHumanReadableName() {
         return "Check loan repayment overdue";
     }
+
+    private static boolean isOverDueEventNeededToBeSent(Loan loan, Long numberOfDaysBeforeDueDateToRaiseEvent, LocalDate currentDate,
+            LoanRepaymentScheduleInstallment repaymentScheduleInstallment, LocalDate repaymentDate) {
+        return repaymentDate.plusDays(numberOfDaysBeforeDueDateToRaiseEvent).equals(currentDate)
+                && loan.getLoanSummary().getTotalOutstanding().compareTo(BigDecimal.ZERO) > 0
```

Comment:
Would you mind to move the `loan.getLoanSummary().getTotalOutstanding().compareTo(BigDecimal.ZERO) > 0` check to the top, next to the loan status check?

Reply:
Done!